mm2src/coins/lightning_persister/src/storage.rs

```diff
@@ -1,33 +1,243 @@
+            funding_value: None,
+            funding_generated_in_block: None,
+            closing_tx: None,
+            closure_reason: None,
+            claiming_tx: None,
+            claimed_balance: None,
+            is_outbound,
+            is_public,
+            is_closed: false,
+            created_at: now_ms() / 1000,
+            last_updated: now_ms() / 1000,
+        }
+    }
+}
+
+#[derive(Clone, Deserialize)]
+pub enum ChannelType {
+    Outbound,
+    Inbound,
+}
+
+#[derive(Clone, Deserialize)]
+pub enum ChannelVisibility {
+    Public,
+    Private,
+}
+
+#[derive(Clone, Deserialize)]
+pub struct ClosedChannelsFilter {
+    pub channel_id: Option<String>,
+    pub counterparty_node_id: Option<String>,
+    pub funding_tx: Option<String>,
+    pub from_funding_value: Option<u64>,
+    pub to_funding_value: Option<u64>,
+    pub closing_tx: Option<String>,
+    pub closure_reason: Option<String>,
+    pub claiming_tx: Option<String>,
+    pub from_claimed_balance: Option<f64>,
+    pub to_claimed_balance: Option<f64>,
+    pub channel_type: Option<ChannelType>,
+    pub channel_visibility: Option<ChannelVisibility>,
+}
+
+pub struct GetClosedChannelsResult {
+    pub channels: Vec<SqlChannelDetails>,
+    pub skipped: usize,
+    pub total: usize,
+}
+
+#[derive(Clone, Debug, Deserialize, Display, PartialEq, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum HTLCStatus {
+    Pending,
+    Succeeded,
+    Failed,
+}
+
+impl FromStr for HTLCStatus {
+    type Err = FromSqlError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "pending" => Ok(HTLCStatus::Pending),
+            "succeeded" => Ok(HTLCStatus::Succeeded),
+            "failed" => Ok(HTLCStatus::Failed),
+            _ => Err(FromSqlError::InvalidType),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum PaymentType {
+    OutboundPayment { destination: Option<PublicKey> },
+    InboundPayment,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct PaymentInfo {
+    pub payment_hash: PaymentHash,
+    pub payment_type: PaymentType,
+    pub description: Option<String>,
+    pub preimage: Option<PaymentPreimage>,
+    pub secret: Option<PaymentSecret>,
+    pub amt_msat: Option<u64>,
+    pub fee_paid_msat: Option<u64>,
+    pub status: HTLCStatus,
+    pub created_at: u64,
+    pub last_updated: u64,
+}
+
+#[derive(Clone)]
+pub struct PaymentsFilter {
+    pub payment_type: Option<PaymentType>,
+    pub description: Option<String>,
+    pub status: Option<HTLCStatus>,
+    pub from_amount_msat: Option<u64>,
+    pub to_amount_msat: Option<u64>,
+    pub from_fee_paid_msat: Option<u64>,
+    pub to_fee_paid_msat: Option<u64>,
+    pub from_timestamp: Option<u64>,
+    pub to_timestamp: Option<u64>,
+}
+
+pub struct GetPaymentsResult {
+    pub payments: Vec<PaymentInfo>,
+    pub skipped: usize,
+    pub total: usize,
+}
+
+#[async_trait]
+pub trait SqlStorage {
```

Comment:
> What you think about grouping the order of function definitions ? I think it would be better in terms of readability and code structure.

I agree. I also shouldn't mix channel-related functions with payment-related functions

Reply:
I also think logical order (order in which the functions are used in the code) is preferred to grouping the add functions together, get functions together, etc.., so in this order for example `fn get_last_channel_rpc_id();` should be before `fn add_channel_to_sql();` since I need to get the last id before adding a new channel with rpc_id equal to last id + 1
What do you think @ozkanonur?

Reply:
> I also think logical order (order in which the functions are used in the code) is preferred to grouping the add functions together, get functions together, etc.., so in this order for example `fn get_last_channel_rpc_id();` should be before `fn add_channel_to_sql();` since I need to get the last id before adding a new channel with rpc_id equal to last id + 1. What do you think @ozkanonur?

Hmm, that sounds cool. But it will be quite hard to review that logical order thing. For example on feature PRs, since we dont' know the implemented logic of that new feature, we might lose some time checking the logical orders are correct or not.

Reply:
For this, I am currently adding a description for every function which should be the norm from now on. These descriptions should make reviewing such logical order easy.

Reply:
> 

Perfect, makes sense then.